src/components/AttendanceChangeModal.tsx

```diff
@@ -0,0 +1,114 @@
+import { XMarkIcon } from "@heroicons/react/24/solid";
+import React, { useState } from "react";
+
+type attendanceChangeProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  //TODO: add a type to this :|
+  setAttendanceChange: any;
+};
+
+const AttendanceChangeModal = ({
+  isOpen,
+  onClose,
+  setAttendanceChange,
+}: attendanceChangeProps) => {
+  // depending on the attendanceChange (give them the option to update arrive_time/leave_time)
+
+  // I want to use the input values from the user typing in their reason for attendanceCHange, but doing this and updating every onChange
+  // seems pretty terrible ...
+  const [reason, setReason] = useState("");
+  const [requestType, setRequestType] = useState("");
+  const [newTime, setNewTime] = useState("");
+
+  const requestOptionHandler = (event: React.FormEvent<HTMLSelectElement>) => {
+    setRequestType(event.currentTarget.value);
+  };
+
+  const availableOptions = [
+    { value: "arrive_late", label: "Arrive Late" },
+    { value: "leave_early", label: "Leave Early" },
+    { value: "absent", label: "Absent" },
+  ];
+
+  const submitForm = () => {
+    //update the variables the json
+    const submissonJson = {
+      reason: reason,
+      request_type: requestType,
+      submission_time: Date.now(),
+    };
+
+    setAttendanceChange({ submissonJson });
+    // close the modal
+    onClose();
+  };
+
+  const resetFields = () => {
```

Comment:
This function is created, but not used. If you could detect when the modal is closed (or after a request is sent), to reset the fields, this would be preferred.